cpp/oneapi/dal/algo/logistic_regression/backend/gpu/train_kernel_dense_batch_dpc.cpp

```diff
@@ -0,0 +1,142 @@
+#include "oneapi/dal/backend/primitives/optimizers.hpp"
+#include "oneapi/dal/algo/logistic_regression/backend/optimizer_impl.hpp"
+
+namespace oneapi::dal::logistic_regression::backend {
+
+using dal::backend::context_gpu;
+
+namespace be = dal::backend;
+namespace pr = be::primitives;
+
+template <typename Float, typename Task>
+static train_result<Task> call_dal_kernel(const context_gpu& ctx,
+                                          const detail::descriptor_base<Task>& desc,
+                                          const detail::train_parameters<Task>& params,
+                                          const table& data,
+                                          const table& resp) {
+    using dal::detail::check_mul_overflow;
+
+    using model_t = model<Task>;
+    using model_impl_t = detail::model_impl<Task>;
+
+    auto opt_impl = detail::get_optimizer_impl(desc);
+
+    if (!opt_impl) {
+        throw internal_error{ dal::detail::error_messages::unknown_optimizer() };
+    }
+
+    auto& queue = ctx.get_queue();
+
+    ONEDAL_PROFILER_TASK(log_reg_train_kernel, queue);
+
+    const auto sample_count = data.get_row_count();
+    const auto feature_count = data.get_column_count();
+    ONEDAL_ASSERT(sample_count == resp.get_row_count());
+    const auto responses_nd =
+        pr::table2ndarray_1d<std::int32_t>(queue, resp, sycl::usm::alloc::device);
+
+    const std::int64_t bsize = params.get_gpu_macro_block();
+
+    const Float L2 = desc.get_l2_coef();
+    const bool fit_intercept = desc.get_compute_intercept();
+
+    // TODO: add check if the dataset can be moved to gpu
+    // Move data to gpu
+    pr::ndarray<Float, 2> data_nd = pr::table2ndarray<Float>(queue, data, sycl::usm::alloc::device);
+    table data_gpu = homogen_table::wrap(data_nd.flatten(queue, {}), sample_count, feature_count);
+
+    pr::logloss_function<Float> loss_func =
+        pr::logloss_function(queue, data_gpu, responses_nd, L2, fit_intercept, bsize);
+
+    auto [x, fill_event] =
+        pr::ndarray<Float, 1>::zeros(queue, { feature_count + 1 }, sycl::usm::alloc::device);
+
+    pr::ndview<Float, 1> x_suf;
+
+    x_suf = fit_intercept ? x : x.slice(1, feature_count);
+
+    auto [train_event, iter_num] = opt_impl->minimize(queue, loss_func, x_suf, { fill_event });
+
+    auto all_coefs = homogen_table::wrap(x.flatten(queue, { train_event }), 1, feature_count + 1);
+
+    const auto model_impl = std::make_shared<model_impl_t>(all_coefs);
+    const auto model = dal::detail::make_private<model_t>(model_impl);
+
+    const auto options = desc.get_result_options();
+    auto result = train_result<Task>().set_model(model).set_result_options(options);
+
+    if (options.test(result_options::intercept)) {
+        ONEDAL_ASSERT(fit_intercept);
+        table intercept_table =
+            homogen_table::wrap(x.slice(0, 1).flatten(queue, { train_event }), 1, 1);
+        result.set_intercept(intercept_table);
```

Comment:
you can use:
` result.set_intercept(homogen_table::wrap(x.slice(0, 1).flatten(queue, { train_event }), 1, 1));`